### PR TITLE
boards: arm: lpcxpresso55s69: Force TF-M when using NS boards

### LINKED ISSUE #35100: libraries.libc.sprintf_new: fails on lpcxpresso55s16_ns and lpcxpresso55s69_ns
**Describe the bug**
 ASSERTION FAI wehn run test_EOF

**To Reproduce**
1. cd tests/lib/sprintf;mkdir build; cd build
2. cmake -DBOARD=lpcxpresso55s16_ns -DCONF_FILE=prj_new.conf ..
3. make; make flash

lpcxpresso55s69_ns has the same problem

**Expected behavior**
test pass

**Impact**
sprint new conf setting features

**Logs and console output**

```
    *** Booting Zephyr OS build zephyr-v2.5.0-3761-g730acbd6ed85  ***
    Running test suite test_sprintf
    ===================================================================
    START - test_EOF
    ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993
    	ESF could not be retrieved successfully. Shall never occur.
    ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993
    	ESF could not be retrieved successfully. Shall never occur.
```

and for cmake -DBOARD=lpcxpresso55s16_ns build we see below failure

```
    *** Booting Zephyr OS build zephyr-v2.5.0-3761-g730acbd6ed85  ***

    Running test suite test_sprintf

    ===================================================================

    START - test_sprintf_double

    ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993

    	ESF could not be retrieved successfully. Shall never occur.

    ASSERTION FAIL [esf != ((void *)0)] @ WEST_TOPDIR/zephyr/arch/arm/core/aarch32/cortex_m/fault.c:993

    	ESF could not be retrieved successfully. Shall never occur.
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v2.6.0-rc1

### PR DESCRIPTION
**Pull request by microbuilder**
This commit forces the `CONFIG_BUILD_WITH_TFM=y` option when using the `_ns` build targets for the LPC55sXX.

Using these targets in samples or in CI without an accompanying secure environment image (ex. TF-M) leads to execution failures, since the NS images are offset a predetermined amount.

Fixes #35100

## Test Output

```bash
$ ./scripts/twister -v --device-testing --build-only -p lpcxpresso55s69_ns -T tests/lib/sprintf
$ JLinkExe -device lpc55s69 -if swd -speed 2000 -autoconnect 1
```

Then in the JLink shell:

```
J-Linke>r
J-Link>erase
J-Link>loadfile twister-out/lpcxpresso55s69_ns/tests/lib/sprintf/libraries.libc.sprintf/tfm_merged.hex 
```

Serial output (post reset):

```
*** Booting Zephyr OS build zephyr-v2.5.0-3928-gfa836cc00d4f  ***
Running test suite test_sprintf
===================================================================
START - test_sprintf_misc
 PASS - test_sprintf_misc in 0.1 seconds
===================================================================
START - test_sprintf_double
 PASS - test_sprintf_double in 0.3 seconds
===================================================================
START - test_sprintf_integer
 PASS - test_sprintf_integer in 0.1 seconds
===================================================================
START - test_vsprintf
 PASS - test_vsprintf in 0.1 seconds
===================================================================
START - test_vsnprintf
 PASS - test_vsnprintf in 0.1 seconds
===================================================================
START - test_sprintf_string
 PASS - test_sprintf_string in 0.1 seconds
===================================================================
START - test_snprintf
 PASS - test_snprintf in 0.1 seconds
===================================================================
START - test_print
3
 PASS - test_print in 0.1 seconds
===================================================================
START - test_fprintf
3
 PASS - test_fprintf in 0.1 seconds
===================================================================
START - test_vfprintf
This 3
        3
/%/a/
11
 PASS - test_vfprintf in 0.3 seconds
===================================================================
START - test_vprintf
This 3
```

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
So only the LPC55S69 seems to have TFM support, so this doesn't make sense.

I still don't quite understand why we have `_ns` variants for 55s28 and 55s16 in zephyr.

**Comment by microbuilder** [comment]
@henrikbrixandersen I'm a bit confused about the LPC55S16 and LPC55S28 having `_ns` targets here since neither of those boards are supported upstream in TF-M. What was the goal adding those, and do you have an out of tree solution for these targets? Just realized after creating this PR that only the LPC55S69 can really pull in TF-M today.

**Comment by microbuilder** [comment]
> So only the LPC55S69 seems to have TFM support, so this doesn't make sense.
> 
> I still don't quite understand why we have `_ns` variants for 55s28 and 55s16 in zephyr.

Correct ... updated this to only force TF-M for the 55s69.

**Comment by henrikbrixandersen** [comment]
@microbuilder I have submitted a PR for renaming the lpcxpresso55s16_ns to drop the _ns part: #35416

**Review by galak — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by ioannisg — CHANGES_REQUESTED** [review]
Does this need any documentation change? E.g. For nordic we write it down that tfm is build by default.

**Comment by microbuilder** [comment]
> Does this need any documentation change? E.g. For nordic we write it down that tfm is build by default.

Good point ... fixed.

**Comment by galak** [comment]
@ioannisg can you take another look

**Review by ioannisg — APPROVED** [review]

File: boards/arm/lpcxpresso55s69/doc/index.rst [hunk 1]
```diff
@@ -92,7 +92,9 @@ CPU0 is the only target that can run standalone.
 - *lpcxpresso55s69_cpu1* CPU1 target, no security extensions
 
 NS target for CPU0 does not work correctly without a secure image enabling it.
-To enable it, run any of the ``tfm_integration`` samples.
+To enable it, run any of the ``tfm_integration`` samples. When using the NS
+target ``CONFIG_BUILD_WITH_TFM`` is enabled to ensure that a valid S image is
+included during the build process.
 
 CPU1 does not work without CPU0 enabling it.
 To enable it, run one of the following samples in ``subsys\ipc``:
```
Comment (mmahadevan108) [inline boards/arm/lpcxpresso55s69/doc/index.rst:100]: The documentation states we should flash tfm_merged.bin. I see only a .hex file, should this be updated to tfm_merged.hex

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**